packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,124 @@
+      'CallExpression[callee.type="Identifier"][callee.name="css"] > ObjectExpression,' +
+      'CallExpression[callee.type="Identifier"][callee.name="cssMap"] > ObjectExpression,' +
+      'CallExpression[callee.type="Identifier"][callee.name="xcss"] > ObjectExpression,' +
+      'CallExpression[callee.type="MemberExpression"][callee.object.name="styled"] > ObjectExpression';
+
+    return {
+      [selectorString]: (node: Rule.Node) => {
+        if (node.type === 'ObjectExpression') {
+          if (node.properties.length > 0) {
+            let lowestDepth = 0;
+            let fixRequired = false;
+            let importedFromCorrectPackage = null;
+            const program = findProgramRoot(node);
+
+            // make sure node is imported from the correct package
+            const logicCssAndCssMap =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'Identifier' &&
+              (node.parent.callee.name === 'css' || node.parent.callee.name === 'cssMap');
+
+            const logicStyled =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'MemberExpression' &&
+              node.parent.callee.object.type === 'Identifier' &&
+              node.parent.callee.object.name === 'styled';
+
+            const logicXcc =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'Identifier' &&
+              node.parent.callee.name === 'xcss';
+
+            if (program && (logicCssAndCssMap || logicStyled)) {
+              importedFromCorrectPackage = program.body.find(
+                (n) => n.type === 'ImportDeclaration' && n.source.value === '@compiled/react'
+              );
+            }
+            if (program && logicXcc) {
+              importedFromCorrectPackage = program.body.find(
+                (n) => n.type === 'ImportDeclaration' && n.source.value === '@atlaskit/primitives'
+              );
+            }
+
+            // loop through the css properties of a ObjectExpression object
+            node.properties.some((property) => {
+              if (property.type === 'Property' && property.key.type === 'Identifier') {
+                const depth = shorthandFor[property.key.name];
+
+                // if we find a property with a with a higher depth below one with a lower depth, we trigger the eslint error
+                if (depth < lowestDepth) {
+                  fixRequired = true;
+                  return;
+                } else {
+                  lowestDepth = depth;
+                }
+              }
+            });
+
+            if (fixRequired && importedFromCorrectPackage) {
+              context.report({
+                node: node,
+                messageId: 'shorthand-first',
+                fix: (fixer) => {
+                  // sort the properties by depth
+                  const sortedProperties = node.properties.slice().sort((a, b) => {
```

Comment:
can we add some tests that include the spread operator ([we lint against this](https://atlassian.design/components/eslint-plugin-design-system/consistent-css-prop-usage/usage) but it is technically possible)

```
const styles = { /* ... */ };
const Component = styled.div({
    ...styles,
    fontColor: 'blue',
});
```